src/System.Net.Security/src/System/Net/Unix/SslConnectionInfo.cs

```diff
@@ -15,30 +15,31 @@ internal class SslConnectionInfo
         public readonly int KeyExchangeAlg;
         public readonly int KeyExchKeySize = 0;
 
-        internal SslConnectionInfo(Interop.libssl.SSL_CIPHER cipher)
+        internal SslConnectionInfo(Interop.libssl.SSL_CIPHER cipher, string protocol)
         {
-            Protocol = (int) MapProtocol(cipher.algorithm_ssl);
+            Protocol = (int) MapProtocolVersion(protocol);
             DataCipherAlg = (int) MapCipherAlgorithmType(cipher.algorithm_enc);
             KeyExchangeAlg = (int) MapExchangeAlgorithmType(cipher.algorithm_mkey);
             DataHashAlg = (int) MapHashAlgorithmType(cipher.algorithm_mac);
             // TODO (Issue #3362) map key sizes
         }
 
-        private static SslProtocols MapProtocol(long algorithm)
+        private SslProtocols MapProtocolVersion(string protocolVersion)
         {
-            switch (algorithm)
+            switch (protocolVersion)
             {
-                case 1:
+                case "SSLv2":
                     return SslProtocols.Ssl2;
-                case 2:
+                case "SSLv3":
                     return SslProtocols.Ssl3;
-                case 3:
+                case "TLSv1":
+                    return SslProtocols.Tls;
+                case "TLSv1.1":
                     return SslProtocols.Tls11;
-                case 4:
+                case "TLSv1.2":
                     return SslProtocols.Tls12;
                 default:
-                    // TODO (Issue #3362) Figure out correct value for this
-                    return SslProtocols.Tls;
+                    return SslProtocols.None;
```

Comment:
I agree with @stephentoub: This would be better as an enum->enum mapping function, instead of a string->enum mapping function.